modules/lidar_detection.py

```diff
@@ -5,39 +5,41 @@
 
 class LidarDetection:
     """
-    Lidar scan
+    Lidar scan containing point relative to drone's position.
     """
 
     __create_key = object()
 
-    __DISTANCE_LIMIT = 50
-    __ANGLE_LIMIT = 170
-
     @classmethod
-    def create(cls, distance: float, angle: float) -> "tuple[bool, LidarDetection | None]":
+    def create(
+        cls, distance: float, angle: float, x: float, y: float
+    ) -> "tuple[bool, LidarDetection | None]":
         """
-        Distance is in meters.
-        Angle is in degrees.
+        distance is in metres.
+        angle is in degrees.
+        (x, y) are the coordinates of the LiDAR scan relative to the drone's position.
+        x is in meters.
+        y is in meters.
         """
-        if distance < 0 or distance > cls.__DISTANCE_LIMIT:
-            return False, None
-
-        if abs(angle) > cls.__ANGLE_LIMIT:
+        if distance == -1:
```

Comment:
why only negative 1? why not keep the old check of 0 <= distance <= limit?

Reply:
the lidar driver takes care of limit checking and returns -1 if it's an invalid read. should I keep checks in both files then?